messaging/src/main/java/org/axonframework/queryhandling/SimpleQueryHandlingComponent.java

```diff
@@ -17,62 +17,106 @@
 package org.axonframework.queryhandling;
 
 import jakarta.annotation.Nonnull;
+import org.axonframework.common.Assert;
+import org.axonframework.common.infra.ComponentDescriptor;
+import org.axonframework.common.infra.DescribableComponent;
 import org.axonframework.messaging.MessageStream;
 import org.axonframework.messaging.QualifiedName;
 import org.axonframework.messaging.unitofwork.ProcessingContext;
 
-import java.util.Objects;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Optional;
 import java.util.Set;
-import java.util.concurrent.ConcurrentHashMap;
 
 /**
- * TODO #3488 This should be regarded as a playground object to verify the API. Feel free to remove, adjust, or replicate this class to your needs.
+ * A simple implementation of the {@link QueryHandlingComponent} interface, allowing for easy registration of
+ * {@link QueryHandler QueryHandlers} and other {@link QueryHandlingComponent QueryHandlingComponents}.
+ * <p>
+ * Registered subcomponents are preferred over registered query handlers when handling a query.
  *
  * @author Steven van Beelen
  * @since 5.0.0
  */
-public class SimpleQueryHandlingComponent
-        implements QueryHandlingComponent, QueryHandlerRegistry<SimpleQueryHandlingComponent> {
+public class SimpleQueryHandlingComponent implements
+        QueryHandlingComponent,
+        QueryHandlerRegistry<SimpleQueryHandlingComponent>,
+        DescribableComponent {
 
-    private final ConcurrentHashMap<QueryHandlerName, QueryHandler> queryHandlers;
+    private final String name;
+    private final Map<QueryHandlerName, QueryHandler> queryHandlers = new HashMap<>();
+    private final Set<QueryHandlingComponent> subComponents = new HashSet<>();
 
-    public SimpleQueryHandlingComponent() {
-        this.queryHandlers = new ConcurrentHashMap<>();
+    /**
+     * Instantiates a simple {@link QueryHandlingComponent} that is able to handle query and delegate them to
+     * subcomponents.
+     *
+     * @param name The name of the component, used for {@link DescribableComponent describing} the component.
+     * @return A simple {@link QueryHandlingComponent} instance with the given {@code name}.
+     */
+    public static SimpleQueryHandlingComponent create(@Nonnull String name) {
+        return new SimpleQueryHandlingComponent(name);
+    }
+
+    private SimpleQueryHandlingComponent(@Nonnull String name) {
+        this.name = Assert.nonEmpty(name, "The name may not be null or empty.");
+    }
+
+    @Override
+    public SimpleQueryHandlingComponent subscribe(@Nonnull QueryHandlerName queryName,
+                                                  @Nonnull QueryHandler handler) {
+        queryHandlers.put(queryName, handler);
```

Comment:
Hmmm... what if I put `QueryHandlingComponent` as an `QueryHandler`? Should it then become `subComponent`? Whats the difference between `subscribe(QueryHandleName, QueryHandler` and `subscribe(QueryHandlingComponent)`? I haven't seen that in other our Message Handling Components and I'm wondering - maybe we should introduce it everywhere? 
In SimpleEventHandler this one looks weird, because I may have handler for different types as a subscriber of some other type....

<img width="1041" height="173" alt="Image" src="https://github.com/user-attachments/assets/bb54d7c1-dde0-4944-9b80-29523e4ee76a" />